### ISSUE TITLE
Feature: Add access keys to context menu items

### ISSUE THREAD

**Issue by adhirajsinghchauhan** [files-community/Files#17759]
### What feature or improvement do you think would benefit Files?

First of all, this is different from #17196 — that only adds access keys to the "toolbar" section of the context menu, for basic operations like cut, copy, paste, rename, share, and options.

This issue is specifically for all the other context menu items, where the mnenomics are usually supplied by third-party apps. For example, see how Windows File Explorer's legacy/full context menu has underlines under some items, which serve as as a shortcut mnemonic to jump directly to that item, which significantly speeds up a user's tasks (we all know that navigating via the keyboard is much faster than the mouse, for most things).

<img width="269" height="609" alt="Image" src="https://github.com/user-attachments/assets/dd6e2d96-0263-47e0-bd2f-332afb40fae3" />

Files does not have this functionality at all. Displaying the underlined letter is not the issue; instead, it's the fact that there are no shortcut/access keys for the rest of the context menu:
<img width="335" height="572" alt="Image" src="https://github.com/user-attachments/assets/0fecda95-743b-4994-b306-9781b1b12a0d" />

For example, tapping the expected shortcut/mnemonic key for Open w<u>i</u>th Code (<kbd>i</kbd>), Open Wit<u>h</u> (<kbd>h</kbd>; for alternative apps that can handle this file), and many more options supplied by other apps, e.g. WinRAR, 7zip, etc., have no effect in Files. What's worse is, the context menu needs to be scrolled through to see all the options, despite there being enough space in the window to show them all at once.

### Requirements

- This proposal will accomplish X
- This proposal will accomplish Y
- This proposal will accomplish Z

### Files Version

4.0.11

### Windows Version

25H2 v26200.6725

### Comments

Keyboard navigability is crucial for power users. It's faster, requires almost no extra work when compared to a mouse (right click, scroll, point, click, each action must be precise, increasing mental load and wasting time).

**Comment by Josh65-2201** [comment]
Thanks for the feedback

**Comment by yair100** [comment]
This appears to be the same as #7505. At the time, we only added the access keys for the most common items (there are limited access keys available). If there is a certain menu item that you feel should have its own access key, please let us know. Otherwise, we recommend using keyboard shortcuts (most actions in Files can be assigned a shortcut from settings) and bypassing the context menu altogether 🙂

**Comment by adhirajsinghchauhan** [comment]
> If there is a certain menu item that you feel should have its own access key

I'm looking for feature parity with Windows File Explorer, i.e. the same access keys that are defined by third-party applications, or by the user by manually editing either of these keys in regedit:
- `HKEY_CLASSES_ROOT\Directory\Background\shell` if admin;
- `HKEY_CURRENT_USER\Software\Classes\directory\shell` otherwise

For example, VS Code already comes with the <kbd>i</kbd> shortcut, as do many other apps (too many to list):
<img width="451" height="18" alt="Image" src="https://github.com/user-attachments/assets/5a259363-a139-4238-98d8-a95297d2409c" />
<img width="405" height="23" alt="Image" src="https://github.com/user-attachments/assets/55a327a8-8861-449f-bf70-99fe28d3403a" />

Users can also manually add an entry into that registry key, and set their own shortcut keys by prefixing a letter with `&`. Windows users are already used to this; pressing the option key on the keyboard to open up the context menu, and then hitting the relevant shortcut key by pure muscle memory, formed over many years of use. It's quick, it's easy, and it's free.

Files does not support these built-in app-provided shortcut keys at all. Your suggestion of assigning our own keyboard shortcuts does not sufficiently solve the problem; for one, why must we reinvent the wheel? Windows already provides such a feature, and apps have been utilizing it without additional configuration or tweaks by the user. **Files should be reading these and assigning the specified shortcut automatically.**

Secondly, any Files-specific keyboard shortcuts will of course require chorded inputs (<kbd>shift</kbd>, <kbd>ctrl</kbd>, <kbd>alt</kbd>, etc.) so that they are less likely to interfere with normal operations, e.g. typeahead to jump to a specific file in a directory listing. Chorded inputs aren't as ergonomic or, simply, easy to use as plain-old single-letter shortcut keys. Moreover, **there may be conflicts with other global shortcuts**, defined in other applications like Game Bar, Discord, NVIDIA App, PowerToys, etc., which would not be easy to discover or fix.

Sticking to an existing, ready-made Windows solution should be considered as the ideal path forward, in my opinion. Files has many great features by itself, built up by the community over the years, but it also lacks what I would consider "default" Windows File Explorer features, that have existed since time immemorial. Wouldn't Files be amazing if new users instantly found their home in it, without needing to re-learn anything? They get the existing good feature-or-two of Windows File Explorer, and they get all the new experiences that Files provides.

**Comment by yair100** [comment]
This feature will be included in the next update.

**Comment by yair100** [comment]
@adhirajsinghchauhan you can now try this in our latest preview build https://files.community/blog/posts/v4-0-41.

**Comment by adhirajsinghchauhan** [comment]
Thank you for the PR. I do confirm that I can see the shortcut keys for the shell extension items. However, there are some alignment issues:
<img width="214" height="136" alt="Image" src="https://github.com/user-attachments/assets/adf37880-83cc-4e46-a81c-5f83733c4bc1" />
<kbd>V</kbd> is supposed to be for Visual Studio, and <kbd>I</kbd> for Visual Studio Code, but with the misalignment it seems like it's for VLC and archive respectively.

Might I recommend copying Windows File Explorer's design here? Simply underline the shortcut/mnemonic. That would avoid this issue entirely. No need to layout and align an overlay. That design language can be kept for Files-specific shortcuts or actions.

---

Moreover…

Upon opening the context menu, the shortcut keys show up only after pressing <kbd>Alt</kbd> first. Also, the newly-added feature in question—access keys for shell extension items—does not show up at the same time. By default, when the context menu opens in Files, the 'actions' row (copy, cut, delete etc.) has focus. So if I press <kbd>Alt</kbd> here, only 'global' Files shortcuts are shown, for the actions row in the context menu, the toolbar (to change sort direction etc.) and the status bar (to open Files Settings).

So, to get the context item access keys to show up, I have to first switch focus into the context menu items by pressing <kbd>Tab</kbd>, and then pressing <kbd>Alt</kbd>. **That's two too many extra keyboard button presses compared to Windows File Explorer.** In WFE, as soon as the context menu opens, you can immediately press the shortcut key to execute the corresponding action. No <kbd>Tab</kbd>/<kbd>Alt</kbd> or any other chord-key required.

In Files, you cannot do that. First, you have to wait for the menu to be populated (I know that you can't do anything about that, it's a limitation of the prettier Windows flyout API), then you have to press two additional buttons, and then finally the shortcut key to execute the action.

So while the PR does technically add the capability to read existing app-defined shortcut keys, it does not surface them to the user in the ideal manner. A few things must be solved, perhaps in a future PR:
1. Remove the need for pressing <kbd>Tab</kbd> first
  When the context menu is open, the focus should already be in that overlay. Global Files shortcuts should only be shown when there's no context menu or overlay open; otherwise, some Files-specific shortcuts may conflict with app-defined ones.
2. Remove the need for pressing <kbd>Alt</kbd> as well, for app-specific shortcuts
  In WFE, you can straightaway go into the execution action by opening the context menu and then pressing the shortcut key. It's how keyboard navigation has worked in pretty much all operating systems, not just Windows. Files requires additional input, which is fine but only if the user can control that behaviour. Some might like it, some don't, and would want to turn it off.

I do appreciate you adding the feature I asked for. If I was familiar with Windows development, I'd have submitted a PR myself. But it's been a long time since I touched this side of software development.
